paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1175 @@
+    return BasicResourceItem(type, name, visibility, configuration, resourceValue)
+  }
+
+  @Throws(XmlSyntaxException::class)
+  private fun getResourceType(
+    tagName: String
+  ): ResourceType? {
+    var type = ResourceType.fromXmlTagName(tagName)
+    if (type == null) {
+      if (TAG_EAT_COMMENT == tagName || TAG_SKIP == tagName) {
+        return null
+      }
+      if (tagName == TAG_ITEM) {
+        val typeAttr = parser.getAttributeValue(null, ATTR_TYPE)
+        if (typeAttr != null) {
+          type = ResourceType.fromClassName(typeAttr)
+          if (type != null) {
+            return type
+          }
+        }
+      }
+    }
+
+    return type
+  }
+
+  /**
+   * If `tagName` is null, calls `subtagVisitor.visitTag()` for every subtag of the current tag.
+   * If `tagName` is not null, calls `subtagVisitor.visitTag()` for every subtag of the current tag
+   * which name doesn't have a prefix and matches `tagName`.
+   */
+  @Throws(
+    IOException::class,
+    XmlPullParserException::class
+  )
+  private fun forSubTags(tagName: String?, subtagVisitor: XmlTagVisitor) {
+    val elementDepth: Int = parser.depth
+    var event: Int
+    do {
+      event = parser.nextToken()
+      if (event == XmlPullParser.START_TAG && (tagName == null || tagName == parser.name && parser.prefix == null)) {
+        subtagVisitor.visitTag()
+      }
+    } while (event != XmlPullParser.END_DOCUMENT && (event != XmlPullParser.END_TAG || parser.getDepth() > elementDepth))
+  }
+
+  @Throws(XmlSyntaxException::class)
+  private fun validateResourceName(
+    resourceName: String,
+    resourceType: ResourceType,
+    file: PathString
+  ) {
+    val error = ValueResourceNameValidator.getErrorText(resourceName, resourceType)
+    if (error != null) {
+      throw XmlSyntaxException(error, parser, file.nativePath)
+    }
+  }
+
+  private fun getDisplayName(file: PathString) = file.nativePath
+
+  private fun getDisplayName(file: File) = getDisplayName(file.toPathString())
+
+  protected fun getVisibility(
+    resourceType: ResourceType,
+    resourceName: String
+  ): ResourceVisibility {
+    // todo look up public resources to return proper visibility
+    return PRIVATE
+    // val names: Set<String> = myPublicResources.get(resourceType)
+    // return if (names != null && names.contains(getKeyForVisibilityLookup(resourceName))) ResourceVisibility.PUBLIC else myDefaultVisibility
+  }
+
+  private fun parseFileResourceFile(
+    file: PathString,
+    folderType: ResourceFolderType,
+    configuration: RepositoryConfiguration
+  ) {
+    val resourceType = FolderTypeRelationship.getNonIdRelatedResourceType(folderType)
+    val resourceName = SdkUtils.fileNameToResourceName(file.fileName)
+    val item = createFileResourceItem(
+      file,
+      resourceName,
+      resourceType,
+      configuration
+    )
+    addResourceItem(item, item.repository as T)
+  }
+
+  private fun getFolderType(file: PathString) =
```

Comment:
nit: inline

Reply:
will be deleted after migrating to new BasicResourceItem